rules/integrations/azure/impact_azure_storage_account_deletion.toml

```diff
@@ -0,0 +1,110 @@
+- Review any related alerts or activities such as data exfiltration, configuration changes, or access policy modifications that occurred before the deletion.
+- Investigate if the account was recently compromised by checking for suspicious authentication events or privilege escalations.
+
+### False positive analysis
+
+- Legitimate decommissioning of unused storage accounts may trigger this alert. Document approved storage account cleanup activities and coordinate with infrastructure teams to understand planned deletions.
+- DevOps automation tools might delete temporary storage accounts as part of infrastructure lifecycle management. Identify service principals used by CI/CD pipelines and consider creating exceptions for these automated processes.
+- Testing and development environments may have frequent storage account creation and deletion cycles. Consider filtering out non-production storage accounts if appropriate for your environment.
+- Cost optimization initiatives may involve deleting unused or redundant storage accounts. Coordinate with finance and infrastructure teams to understand planned resource optimization activities.
+
+### Response and remediation
+
+- Immediately investigate whether the deletion was authorized by verifying with the account owner or relevant stakeholders.
+- If the deletion was unauthorized, attempt to recover the storage account if soft-delete is enabled, or restore data from backups.
+- Disable the compromised user account or service principal if unauthorized activity is confirmed and investigate how the credentials were obtained.
+- Review and restrict Azure RBAC permissions to ensure only authorized users have storage account deletion capabilities (requires Contributor or Owner role).
+- Implement Azure Resource Locks to prevent accidental or malicious deletion of critical storage accounts.
+- Configure Azure Activity Log alerts to notify security teams immediately when storage accounts are deleted.
+- Conduct a full security assessment to identify any other compromised resources or accounts and look for indicators of broader compromise.
+- Document the incident and update security policies and procedures to prevent similar incidents in the future.
+"""
+references = [
+    "https://www.microsoft.com/en-us/security/blog/2025/08/27/storm-0501s-evolving-techniques-lead-to-cloud-based-ransomware/"
+]
+risk_score = 47
+rule_id = "a1b2c3d4-5e6f-7a8b-9c0d-1e2f3a4b5c6d"
+severity = "medium"
+tags = [
+    "Domain: Cloud",
+    "Domain: Storage",
+    "Data Source: Azure",
+    "Data Source: Azure Activity Logs",
+    "Use Case: Threat Detection",
+    "Tactic: Impact",
+    "Resources: Investigation Guide",
+]
+timestamp_override = "event.ingested"
+type = "new_terms"
+
+query = '''
+event.dataset: azure.activitylogs and
+    azure.activitylogs.operation_name: "MICROSOFT.STORAGE/STORAGEACCOUNTS/DELETE" and
+    azure.activitylogs.identity.claims_initiated_by_user.name: *
```

Comment:
is `azure.activitylogs.identity.claims_initiated_by_user.name` field ever not populated?